sqlglot/dialects/clickhouse.py

```diff
@@ -629,11 +628,23 @@ def _parse_bracket(
 
             return bracket
 
-        def _parse_in(self, this: t.Optional[exp.Expression], is_global: bool = False) -> exp.In:
+        def _parse_in(
+            self,
+            this: t.Optional[exp.Expression],
+            is_global: bool = False,
+            is_negated: bool = False,
+        ) -> exp.In:
             this = super()._parse_in(this)
             this.set("is_global", is_global)
+            this.set("is_negated", is_negated)
```

Comment:
This isn't correct, we shouldn't be setting args that don't appear in `arg_types`.

I'm not sure what the best approach is here... Perhaps we can normalize this into: `Not(In(..., is_global=True))` and then override ClickHouse's generator methods to properly generate this.

Reply:
Thanks, I switched to that & tried `str.replace(..., 1)`. Is there a saner way to replace the syntax?

Reply:
Looks sane enough to me :)

Thanks for addressing this!